### ISSUE TITLE
1.1.4 does not quote time ranges

### ISSUE THREAD

**Issue by jklimke** [palkan/influxer#37]
Due to the changes done for version 1.1.4, time ranges are not escaped anymore in where clauses.

New quotation code:

```
def quoted(val, key = nil)
      if val.is_a?(String) || val.is_a?(Symbol) || @klass.tag?(key)
        "'#{val}'"
      elsif key.to_s == "time"
        quote_timestamp val, @instance.client
      else
        val.to_s
      end
    end
```

The code before checked if the value was a `Time`, `DateTime`, or `Date` instance and applied timestamp quoting appropriately:

```
  def quoted(val, key = nil)
       if val.is_a?(String) || val.is_a?(Symbol) || @klass.tag?(key)
         "'#{val}'"
-      elsif val.is_a?(Time) || val.is_a?(Date) || val.is_a?(DateTime)
+      elsif key.to_s == "time"
         quote_timestamp val, @instance.client
       else
         val.to_s
       end
     end
```

The solution would be either to pass the key with every call to `quoted` or to reintroduce time quoting by instance type

**Comment by jklimke** [comment]
By the way... does it actually make sensor to check for keys named 'time' only for escaping dates, since other values in a influx table may also have the time type ?

**Comment by palkan** [comment]
>  does it actually make sensor to check for keys named 'time' only for escaping dates, since other values in a influx table may also have the time type ?

As I see, there is no first-class support for time types in InfluxDB https://docs.influxdata.com/influxdb/v1.4/concepts/glossary/#field-value. So, it's up to developers to use the correct representation (we don't know whether you store time as integers or strings).

Only `time` has specific meaning.

**Comment by jklimke** [comment]
just added the changelog, thanks for the pointer to the field values

**Comment by palkan** [comment]
Released as 1.1.5
